app/jobs/promoted_click_accounter_job.rb

```diff
@@ -0,0 +1,11 @@
+class PromotedClickAccounterJob < ApplicationJob
+  queue_as :default
+
+  def perform(product_action, organization)
+    PromotedClickAccounter.for(
+      organization: organization,
+      store: product_action.product.store,
```

Comment:
Podrías usar delegate para no acceder a store a través de product
https://devblast.com/b/rails-delegate-dont-break-the-law-of-demeter